storage.lua

```diff
@@ -135,8 +173,53 @@ function mail.delete_mail(playername, msg_ids)
 			end
 		end
 	end
+	if delete_in_trash then
+		for i = #entry.trash, 1, -1 do
+			for _, deleted_msg in ipairs(msg_ids) do
+				if entry.trash[i].id == deleted_msg then
+					table.remove(entry.trash, i)
```

Comment:
needs a `break` after that, otherwise it may trigger the issue we had in #97 again :smirk:

Reply:
Done